runtime/lua/vim/treesitter/query.lua

```diff
@@ -328,24 +336,26 @@ local directive_handlers = {
   -- Example: (#offset! @_node 0 1 0 -1)
   ['offset!'] = function(match, _, _, pred, metadata)
     local capture_id = pred[2]
-    local offset_node = match[capture_id]
-    local range = { offset_node:range() }
-    local start_row_offset = pred[3] or 0
-    local start_col_offset = pred[4] or 0
-    local end_row_offset = pred[5] or 0
-    local end_col_offset = pred[6] or 0
-
-    range[1] = range[1] + start_row_offset
-    range[2] = range[2] + start_col_offset
-    range[3] = range[3] + end_row_offset
-    range[4] = range[4] + end_col_offset
-
-    -- If this produces an invalid range, we just skip it.
-    if range[1] < range[3] or (range[1] == range[3] and range[2] <= range[4]) then
-      if not metadata[capture_id] then
-        metadata[capture_id] = {}
+    for _, offset_node in ipairs(match[capture_id]) do
+      local range = { offset_node:range() }
+      local start_row_offset = pred[3] or 0
+      local start_col_offset = pred[4] or 0
+      local end_row_offset = pred[5] or 0
+      local end_col_offset = pred[6] or 0
+
+      range[1] = range[1] + start_row_offset
+      range[2] = range[2] + start_col_offset
+      range[3] = range[3] + end_row_offset
+      range[4] = range[4] + end_col_offset
+
+      -- If this produces an invalid range, we just skip it.
+      if range[1] < range[3] or (range[1] == range[3] and range[2] <= range[4]) then
+        if not metadata[capture_id] then
+          metadata[capture_id] = {}
+        end
+        -- TODO: change metadata[capture_id] to support multiple nodes per capture
+        metadata[capture_id].range = range
```

Comment:
This looks broken. If there are multiple nodes then range gets overwritten on each iteration.